mkfs/rootdir.c

```diff
@@ -74,6 +75,50 @@ struct inode_entry {
 	struct list_head list;
 };
 
+/*
+ * Record all the hard links we found for a specific file inside
+ * rootdir.
+ *
+ * The search is based on (root, st_dev, st_ino).
+ * The reason for @root as a search index is, for hard links separated by
+ * subvolume boundaries:
+ *
+ * rootdir/
+ * |- foobar_hardlink1
+ * |- foobar_hardlink2
+ * |- subv/	<- Will be created as a subvolume
+ *    |- foobar_hardlink3.
+ *
+ * Since all the 3 hard links are inside the same rootdir and the same
+ * filesystem, on the host fs they are all hard links to the same inode.
+ *
+ * But for created btrfs, only hardlink1 and hardlink2 can be created as
```

Comment:
Please rephrase "for created btrfs", I'm not sure I understand what you mean.

Reply:
I changed it to "But for the btrfs we are building".